packages/block-library/src/template-part/edit/selection/template-part-previews.js

```diff
@@ -246,18 +262,24 @@ export default function TemplatePartPreviews( {
 	filterValue,
 	onClose,
 	area,
+	templatePartId,
 } ) {
 	const composite = useCompositeState();
 
 	const { templateParts, labelsByArea } = useSelect( ( select ) => {
-		const _templateParts =
+		const _templateParts = (
 			select( coreStore ).getEntityRecords(
 				'postType',
 				'wp_template_part',
 				{
 					per_page: -1,
 				}
-			) || [];
+			) || []
+		).filter(
+			( templatePart ) =>
+				`${ templatePart.theme }//${ templatePart.slug }` !==
+				templatePartId
```

Comment:
`unrelated`: We really should have a util function for creating a `templatePartId` 😬

Reply:
Good point.  I just added one in `./utils` 😁